tests/suites/storage/model_storage.sh

```diff
@@ -0,0 +1,143 @@
+	wait_for_storage "attached" '.storage["single-fs/0"]["status"].current'
+	assert_storage "filesystem" "$(kind_name "single-fs" 0)"
+	assert_storage "alive" "$(life_status "single-fs" 0)"
+	assert_storage "dummy-storage/0" "$(unit_attachment "single-fs" 0 0)"
+
+	echo "Filesystem storage type with tmpfs test PASSED"
+	# Clean up
+	juju remove-application -m "${model_name}" --force --no-prompt dummy-storage
+	wait_for "{}" ".applications"
+
+	echo "Testing model config filesystem storage type with rootfs"
+
+	# Set default filesystem storage type in model config
+	juju model-config -m "${model_name}" storage-default-filesystem-source=rootfs
+
+	# Deploy application with filesystem storage without specifying storage type
+	juju deploy -m "${model_name}" ./testcharms/charms/dummy-storage --storage single-fs=1G
+
+	# Wait for the application to be active
+	wait_for "dummy-storage" "$(active_condition "dummy-storage" 0)"
+	wait_for "active" "$(workload_status "dummy-storage" 1).current"
+
+	# Verify rootfs storage details
+	wait_for_storage "attached" '.storage["single-fs/1"]["status"].current'
+	assert_storage "filesystem" "$(kind_name "single-fs" 1)"
+	assert_storage "alive" "$(life_status "single-fs" 1)"
+	assert_storage "dummy-storage/1" "$(unit_attachment "single-fs" 1 0)"
+
+	echo "Filesystem storage type with rootfs test PASSED"
+
+	# Clean up
+	juju remove-application -m "${model_name}" --force --no-prompt dummy-storage
+	wait_for "{}" ".applications"
+
+	echo "All model config filesystem storage tests PASSED"
+}
+
+# Generic test for block storage using the provided storage type
+test_block_storage() {
+	local model_name=$1
+	local storage_type=$2
+
+	echo "Testing model config block storage type with ${storage_type}"
+
+	# Set default block storage type in model config
+	juju model-config -m "${model_name}" storage-default-block-source=${storage_type}
+
+	# Deploy application with block storage without specifying storage type
+	juju deploy -m "${model_name}" ./testcharms/charms/dummy-storage --storage single-blk=1G
+
+	# Wait for the application to be active
+	wait_for "dummy-storage" "$(active_condition "dummy-storage" 0)"
+	wait_for "active" "$(workload_status "dummy-storage" 2).current"
+
+	# Verify storage details
+	wait_for_storage "attached" '.storage["single-blk/2"]["status"].current'
+	assert_storage "block" "$(kind_name "single-blk" 2)"
+	assert_storage "alive" "$(life_status "single-blk" 2)"
+	assert_storage "dummy-storage/2" "$(unit_attachment "single-blk" 2 0)"
+
+	echo "Block storage test with ${storage_type} PASSED"
+
+	# Clean up
+	juju remove-application -m "${model_name}" --force --no-prompt dummy-storage
+	wait_for "{}" ".applications"
+}
+
+# Main test function that runs all the subtests
+test_model_storage() {
+	if [ "$(skip 'test_model_storage')" ]; then
+		echo "==> TEST SKIPPED: model storage type tests"
+		return
+	fi
+
+	(
+		set_verbosity
+		cd .. || exit
+
+		# Detect environment and run appropriate tests
```

Comment:
The detection is not being done here - we're just echoing out the provider name for debug
We can drop this here and do the echo in the place where we determine what type of storage etc

Reply:
fixed.